src/AWS/CRT/HTTP/Message.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace AWS\CRT\HTTP;
+
+abstract class Message {
+    private $path = "";
+    private $query = [];
+    private $headers = [];
+
+    public function __construct($path, $query = [], $headers = []) {
+        $this->path = $path;
+        $this->query = $query;
+        $this->headers = $headers;
+    }
+}
+
+class Request extends Message {
```

Comment:
Normally we would have these in two different files, so that if we make another extending class it is better organized

Reply:
Will do in follow up where I actually use these classes!